internal/db/migrate.go

```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-func Migrate(connStr string, migrationsDir string, logger *logrus.Logger) error {
+func Migrate(connStr string, migrationsDir string, steps int, logger *logrus.Logger) error {
```

Comment:
Maybe we can do it in a way that leaves the compatibility with the previous API of migrate?

`func Migrate(connStr string, migrationsDir string, logger *logrus.Logger) error {` does the full migration
`func MigrateStep(connStr string, migrationsDir string, steps int, logger *logrus.Logger) error {` executes the given step for the migration. With an precodition on steps being > 0.

This way we can avoid the Hack of the step 0.

Another solution, could be to make `Migrate` invoke a callback function between each step.

What are your thoughts ?

Reply:
I'll add a 2nd function, I think that'll make it the easiest to read as well.